tests/unit/Command/RecreateMasterKeyTest.php

```diff
@@ -134,106 +99,83 @@ public function setUp() {
-				->willReturn(true);
-
-			$this->util->expects($this->any())->method('isMasterKeyEnabled')
-				->willReturn(true);
-
-			$this->userManager->expects($this->any())
-				->method('getBackends')
-				->willReturn([$this->userInterface]);
-
-			$this->userInterface->expects($this->any())
-				->method('getUsers')
-				->willReturn(['user1']);
-
-			$this->rootView->expects($this->any())->method('is_dir')
-				->willReturnCallback(
-					function ($path) {
-						if ($path === '/user1/files/foo') {
-							return true;
-						}
-						return false;
-					}
-				);
-
-			$this->encryptionManager->expects($this->any())
-				->method('isReadyForUser')
-				->with('user1')
-				->willReturn(true);
-
-			$this->output->method('writeln')
-				->will($this->onConsecutiveCalls(
-					"Decryption started\n",
-					"\nDecryption completed\n",
-					"Encryption started\n",
-					"Waiting for creating new masterkey\n",
-					"New masterkey created successfully\n",
-					"\nEncryption completed successfully\n",
-					"\n\<info\>Note: All users are required to relogin.\</info\>\n"
-				));
-			$this->invokePrivate($this->recreateMasterKey, 'execute', [$this->input, $this->output]);
-		} else {
-			$this->recreateMasterKey = $this->getMockBuilder('OCA\Encryption\Command\RecreateMasterKey')
-				->setConstructorArgs(
-					[
-						$this->userManager,
-						$this->rootView, $this->keyManager, $this->util, $this->encUitl,
-						$this->IAppManager, $this->appConfig, $this->config,
-						$this->session, $this->encryptionManager, $this->questionHelper,
-						$this->userSetup, $this->mailer, $this->secureRandom, $this->l,
-						$this->logger
-					]
-				)->setMethods(['setupUserFS'])->getMock();
-
-			$this->util->expects($this->once())->method('isMasterKeyEnabled')
-				->willReturn($mastkerKeyEnabled);
-
-			global $outputText;
-			$this->output->expects($this->at(0))
-				->method('writeln')
-				->willReturnCallback(function ($value) {
-					global $outputText;
-					$outputText .= $value . "\n";
-				});
-
-			$this->invokePrivate($this->recreateMasterKey, 'execute', [$this->input, $this->output]);
-			$this->assertSame("Master key is not enabled.", \trim($outputText, "\n"));
-		}
+	public function testRecreateMasterKey() {
+		$this->util->method('isMasterKeyEnabled')
+			->willReturn(true);
+
+		$decryptAll = $this->createMock(DecryptAll::class);
+		$decryptAll->expects($this->once())
+			->method('decryptAll');
+
+		$encryptAll = $this->createMock(EncryptAll::class);
+		$encryptAll->method('createMasterKey')
+			->willReturn(true);
+		$encryptAll->expects($this->once())
+			->method('encryptAll');
+
+		$this->encdecAllFactory->method('getDecryptAllObj')
+			->willReturn($decryptAll);
+		$this->encdecAllFactory->method('getEncryptAllObj')
+			->willReturn($encryptAll);
+
+		$this->commandTester->execute([
+			'--yes' => 'y'
+		]);
+
+		$output = $this->commandTester->getDisplay();
+		$expectedOutput = "Decryption completed
```

Comment:
I'd rather check the exit code of the command instead. You can use different constants to check for different errors.

Reply:
Added return values to the command. I know its better to check with the exit code. Somehow I would like an additional assert with output strings.